### DISCUSSION TITLE
Is possible configure the timeout limit?

### DISCUSSION THREAD

**Discussion by wwwflorencio** [vercel/vercel#4502] [Help] [answered]
I have a function with a long process to convert some audio/video files with [ffmpeg](https://ffmpeg.org/) and this process time is greater than 10ms

![image](https://user-images.githubusercontent.com/5944629/83032838-b7af6b80-a00c-11ea-9429-6190a6df6131.png)

The project is a telegram bot, you can test [here](https://t.me/djmarcinhobot)

**Comment by styfle** [comment]
Yes, you can configure the `maxDuration` using the [`functions`](https://vercel.com/docs/configuration#project/functions) property.

Here's an example `vercel.json`:

```json
{
  "functions": {
    "api/test.js": {
      "maxDuration": 60
    }
  }
}
```

Note that anything longer than 10 seconds requires paid plan.

https://vercel.com/docs/v2/platform/limits

**Comment by NachoJusticia** [comment]
I am having some problems when setting up a `maxDuration` to the maximum that my plan accepts, PRO tier, 60seconds.
I put this in the vercel.json:
```
	"functions": {
		"api/test.ts": {
			"maxDuration": 60
		}
	}
```

I am getting this error:
```
 vercel dev
Vercel CLI 19.0.1 dev (beta) — https://vercel.com/feedback
Error! The function for api/test.ts can't be handled by any builder
```

I have also tried to add the runtime, nodeJS, which I think is also optional, like that:
```
	"functions": {
		"api/test.ts": {
			"maxDuration": 60,
			"runtime": "now-node@1.6.1"
		}
	}
```
but I got the same error.

**Reply by styfle** [reply]
If you're using Next.js, then you will need to use `pages/api/**/*.ts` like the following:

```json
{
  "functions": {
    "pages/api/**/*.ts": {
      "maxDuration": 60
    }
  }
}
```

The glob should match the source files.

**Reply by ZainAli104** [reply]
I am using node js so which route I can define as I have this route:
localhost:3000/api/v1/dalle

so what I can write in vercel.json

**Comment by submetu** [comment]
Hi, Does anyone know what to do in case of Remix? 
`api/**/*.js` or `api/index.js` matches don't work :(

**Reply by braedengoodwin** [reply]
Not sure if you solved this but this did work for my remix project. Strange that it didn't work for yours

{
    "functions": {
      "api/**/*.ts": {
        "maxDuration": 60
      }
    }
  }

**Comment by Nutlope** [comment, marked as answer]
The timeout limit for serverless functions on the Hobby tier is 10 seconds and it's 60 seconds on Pro. However, edge functions have a longer timeout and are available on all plans so I recommend you check those out to see if they'll work for your use case.

You can get started with edge functions here: https://vercel.com/docs/concepts/functions/edge-functions/quickstart

Also, if you're using AI APIs, you can learn how to do streaming with edge functions here: https://vercel.com/blog/gpt-3-app-next-js-vercel-edge-functions

**Reply by ShantanuNair** [reply]
@Nutlope Can you confirm what exactly the longer timeout limit for edge functions are? I know the CPU Time's limit, but am curious if there is a wall clock limit.
